### ISSUE TITLE
afterAll callback not called when using preload script

### ISSUE THREAD

**Issue by IlyaSemenov** [oven-sh/bun#4035]
### What version of Bun is running?

0.7.2

### What platform is your computer?

Darwin 22.5.0 arm64 arm

### What steps can reproduce the bug?

```sh
bun add orchid-orm pqm
createdb test
```

Test file `foo.test.ts`:

```ts
import { test } from "bun:test"

import { db } from "./setup"

test("test", async () => {
	await db.$query`select 1`
})
```

Preload file `setup.ts`:

```ts
import { afterAll, beforeAll } from "bun:test"
import { orchidORM, testTransaction } from "orchid-orm"

export const db = orchidORM({ databaseURL: "postgres://user@localhost/test", log: true }, {})

beforeAll(async () => {
	console.log("before all")
	await testTransaction.start(db)
})

afterAll(async () => {
	console.log("after all")
	await testTransaction.close(db)
})
```

Run it with: `bun test -r ./setup.ts foo.test.ts`

### What is the expected behavior?

The test passes. `afterAll` is called and at least prints to console.

### What do you see instead?

`afterAll` is never called, the test runner hangs (probably due to open pg pool with active transaction):

```
bun test v0.7.2 (9574db35)

foo.test.ts:
before all
(9.8ms) BEGIN
(0.5ms) select 1
✓ test [0.74ms]
<--- here it hangs forever
```

### Additional information

Everything works if the test and beforeAll/afterAll are in the single file `foo.test.ts`:

```ts
import { afterAll, beforeAll, test } from "bun:test"
import { orchidORM, testTransaction } from "orchid-orm"

const db = orchidORM({ databaseURL: "postgres://user@localhost/test", log: true }, {})

beforeAll(async () => {
	console.log("before all")
	await testTransaction.start(db)
})

afterAll(async () => {
	console.log("after all")
	await testTransaction.close(db)
})

test("test", async () => {
	await db.$query`select 1`
})
```

```
bun test v0.7.2 (9574db35)

foo.test.ts:
before all
(22.5ms) BEGIN
(1.5ms) select 1
✓ test [1.72ms]
after all
(3.9ms) ROLLBACK

 1 pass
 0 fail
Ran 1 tests across 1 files. [89.00ms]
```

**Comment by timurguvenkaya** [comment]
I'm experiencing exactly the same issue 

afterAll is not called and test hangs

**Comment by timurguvenkaya** [comment]
I had this issue 

Bun reaches global afterAll after all tests are done. If there is still an open connection pending it won't reach afterAll until that connection is closed

So I changed my global setup like this and it worked

```typescript
await prepareDbTestData().then(() => {
	console.debug('DB prepared');
});

afterEach(async () => {
	try {
		const connection = await global.connectionPool.getConnection();
		connection.destroy();
	} catch (_) {
		console.debug('Connection already closed');
	}
});

afterAll(async () => {
	console.debug('DB cleaned');
	await cleanDb();
});
```

On my side beforeAll is late. If I put prepareDb code inside beforeAll, beforeAll runs later than my first test causing DB error..

**Comment by Th1nkK1D** [comment]
Found the same issue here on Bun `1.0.11` with both [postgres.js](https://github.com/porsager/postgres) and [pg](https://github.com/brianc/node-postgres). Likely it has something to do with the Postgres connection that is still alive. A similar issue and workaround were reported to https://github.com/porsager/postgres/issues/682

**Comment by masterT** [comment]
This seems fixed with Bun `1.0.13`.

**Comment by Th1nkK1D** [comment]
> This seems fixed with Bun `1.0.13`.

Confirmed. Upgraded to `1.0.13` and the issue is fixed.
